pkg/kubecost/allocation.go

```diff
@@ -1135,7 +1147,11 @@ func (as *AllocationSet) AggregateBy(aggregateBy []string, options *AllocationAg
 	// an empty slice implies that we should aggregate everything. See
 	// generateKey for why that makes sense.
 	shouldAggregate := aggregateBy != nil
-	shouldFilter := options.Filter != nil
+	shouldFilter := true
+	// TODO: Make sure this works as expected
```

Comment:
Still TO DO?

Reply:
Is this potentially something we could/should add to the interface? Something like `IsEmpty()`?

Reply:
This was a poke I needed! Thank you. I'm wary of expanding the interface for logic that is only used by AggBy, so I moved this logic into a private helper func with unit tests.